src/state/State.ts

```diff
@@ -216,6 +248,21 @@ export class State {
         }
     }
 
+    chercheCommuneParCode(basePath: string, codePostal: string, codeCommune: string): Promise<Commune> {
+        return this.communeAutocompleteTriggers(basePath)
```

Comment:
@nhumblot je ne suis pas sûr que tu aies besoin de passer par la case `communeAutocompleteTriggers()`

À mon avis, si tu appelles directement le `communesPourAutocomplete(basePath, codePostal)` ça devrait fonctionner (et simplifie un peu le code)

PS: n'hésites pas à utiliser async/await ;-)

Reply:
Si je change l'implémentation de `chercheCommuneParCode()` par :

```
return this.communesPourAutocomplete(basePath, codePostal)
    .then(communes => communes.filter(commune => commune.code === codeCommune))
    .then(communesFiltrees => communesFiltrees.length > 0 ? communesFiltrees[0] : State.COMMUNE_VIDE);
```

Une recherche sur "21000 - Dijon" génère une erreur avec un appel à `GET http://localhost:3000/autocomplete-cache/21000.json 404 (Not Found)`

J'ai observé le comportement de la recherche, si j'effectue une recherche en tapant progressivement "Dijon", l'autocomplete va envoyer seulement "di" en argument à la méthode `communesPourAutocomplete()` qui va renvoyer "di.json". `communesPourAutocomplete()` prend l'entrée telle quelle et cherche un fichier json avec le même nom.

Ou alors tu veux que la méthode `communesPourAutocomplete()` se mette à gérer ces cas ? Autrement dit, tu souhaites que son implémentation commence par récupérer les triggers pour sélectionner celui qui convient comme réalisé dans la méthode `chercheCommuneParCode()` avec quelque chose comme :
```
let trigger = await this.communeAutocompleteTriggers(basePath)
    .filter(trigger => autocomplete.startsWith(trigger)))[0]
if(this._communesParAutocomplete.has(trigger)) {
// ...
```
?

Reply:
J'ai ajouté les `async / await` pour clarifier. :+1: 

Je n'ai pas modifié l'implémentation (cf: commentaire plus haut). Je préfère attendre ton retour pour savoir quelle direction on prend dessus.

Reply:
> Une recherche sur "21000 - Dijon" génère une erreur avec un appel à GET http://localhost:3000/autocomplete-cache/21000.json 404 (Not Found)

Oui tu as raison, ne change rien c'est moi qui raconte n'importe quoi : on ne peut pas faire beaucoup plus simple que ce que tu as fait.